RHEL 10 STIG Update

Mab879 opened the pull request:
#### Description:

* Update RHEL 10 STIG
* Update to V3R1 SRG GPOS
#### Rationale:

* Prep for RHEL 10

#### Review Hints:
```
./build_product rhel10
cd tests
./ds_unselect_rules.sh ../build/ssg-rhel10-ds.xml unselect_rules_list
./automatus.py profile --libvirt qemu:///system automatus_rhel10 --datastream /tmp/ssg-rhel10-ds.xml stig
```

marcusburghardt commented on controls/srg_gpos/SRG-OS-000590-GPOS-00110.yml:
```diff
@@ -0,0 +1,8 @@
+        levels:
+            - medium
+        status: pending
+
```
Nitpick: two empty lines.

marcusburghardt commented on controls/srg_gpos/SRG-OS-000725-GPOS-00180.yml:
```diff
@@ -0,0 +1,8 @@
+        levels:
+            - medium
+        status: pending
+
```
Nitpick: two empty lines.

marcusburghardt commented on controls/srg_gpos/SRG-OS-000730-GPOS-00190.yml:
```diff
@@ -0,0 +1,8 @@
+        levels:
+            - medium
+        status: pending
+
```
Nitpick: two empty lines.

marcusburghardt commented on controls/srg_gpos/SRG-OS-000745-GPOS-00210.yml:
```diff
@@ -0,0 +1,7 @@
+        levels:
+            - medium
+        status: pending
+
```
Nitpick: two empty lines.

marcusburghardt commented on controls/srg_gpos/SRG-OS-000775-GPOS-00230.yml:
```diff
@@ -0,0 +1,8 @@
+        levels:
+            - medium
+        status: pending
+
```
Nitpick: two empty lines.

marcusburghardt commented on controls/srg_gpos/SRG-OS-000780-GPOS-00240.yml:
```diff
@@ -0,0 +1,8 @@
+        levels:
+            - medium
+        status: pending
+
```
Nitpick: two empty lines.

marcusburghardt commented on linux_os/guide/system/software/updating/dnf-automatic_apply_updates/policy/stig/shared.yml:
```diff
@@ -0,0 +1,25 @@
+srg_requirement: |-
+    {{{ full_name }}} must configure dnf-automatic to Install Available Updates Automatically.
+
```
Nitpick: two empty lines.

marcusburghardt reviewed (changes requested):
The updates looks good to me. I only have some minor considerations regarding consistence, such as empty lines and quotes. I also saw that some rules are using `vuldiscussion` in their stig policy declaration while others use `vuln_discussion`. Are both accepted? It would be good to keep a consistence on this too.

marcusburghardt commented:
```
./automatus.py profile --libvirt qemu:///session rhel10 --datastream /tmp/ssg-rhel10-ds.xml stig
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Evaluation of the profile has passed: xccdf_org.ssgproject.content_profile_stig (initial stage).
INFO - Evaluation of the profile has passed: xccdf_org.ssgproject.content_profile_stig (remediation stage).
INFO - Rebooting domain 'rhel10' before final scan.
INFO - Waiting for 30 seconds to let the system finish startup.
INFO - Evaluation of the profile has passed: xccdf_org.ssgproject.content_profile_stig (final stage).
```

ggbecker commented on .github/workflows/srg-mapping-table.yaml:
```diff
@@ -44,20 +44,20 @@ jobs:
           PYTHONPATH: ${{ github.workspace }}
-      - name: Generate XLSX for RHEL9
-        run: python3 utils/create_srg_export.py -c controls/srg_gpos.yml -p rhel9 -m shared/references/disa-os-srg-v2r7.xml --out-format xlsx --output $PAGES_DIR/srg-mapping-rhel9.xlsx
+      - name: Generate XLSX for RHEL9disa-os-srg-v3r1.xml
```
typo here?

Mab879 commented:
> The updates looks good to me. I only have some minor considerations regarding consistence, such as empty lines and quotes. I also saw that some rules are using `vuldiscussion` in their stig policy declaration while others use `vuln_discussion`. Are both accepted? It would be good to keep a consistence on this too.

Thanks for the review, I have cleaned up titles and empty lines.

I cleaned up the `vuln_discussion` and fixed the bug the import script.

I also remove the extraneous text in the GitHub Actions file.

marcusburghardt reviewed (approved):
LGTM. There is a lot of changes related to text. They were automated and shouldn't have any technical impact, but it would be good other eyes to also take a look if I missed any typo. @ggbecker , could you also take a look, please?

ggbecker commented on linux_os/guide/services/mail/package_s-nail_installed/policy/stig/shared.yml:
```diff
@@ -21,3 +21,6 @@ checktext: |-
 vuln_discussion: |-
     The "s-nail" package provides the mail command required to allow sending email notifications of unauthorized configuration changes to designated personnel.
+
+vuldiscussion: |-
```
I'm confused why there is `vuln_discussion` vs `vuldiscussion`, I think we should unify them into a single field or is there a reason for that?

ggbecker replied:
There are many occurrences of both in the project

marcusburghardt replied:
Yes. I agree we should unify them. But I believe this could also be solved in a separate PR, to make the review easier.

ggbecker reviewed (changes requested):
Revert back the {{{ full_name }}} in various file. I think you do a search through all the stig files with `Red Hat Enterprise Linux 9` to find where we need to revert the change.

Regarding the vuldiscussion vs vuln_discussion topic, I still see places with the vuln_discussion field. It's worth to double check that because there seems to have duplicated data there.

I see that many of the changes are only formatting options, if we can't avoid that, I suggest we do that in a different pull request... maybe having the script to detect if the text is actually different and only perform the change if there is any. Then as a follow up we can open another pull request that will just perform this type of formatting change.

Mab879 commented:
> Revert back the {{{ full_name }}} in various file. I think you do a search through all the stig files with `Red Hat Enterprise Linux 9` to find where we need to revert the change.
> 
> Regarding the vuldiscussion vs vuln_discussion topic, I still see places with the vuln_discussion field. It's worth to double check that because there seems to have duplicated data there.
> 
> I see that many of the changes are only formatting options, if we can't avoid that, I suggest we do that in a different pull request... maybe having the script to detect if the text is actually different and only perform the change if there is any. Then as a follow up we can open another pull request that will just perform this type of formatting change.

For how I have decided that addressing the `vuldiscussion` vs `vuln_discussion` should be reserved for another PR. So I have dropped those commits from the PR.

ggbecker commented:
> > Revert back the {{{ full_name }}} in various file. I think you do a search through all the stig files with `Red Hat Enterprise Linux 9` to find where we need to revert the change.
> > Regarding the vuldiscussion vs vuln_discussion topic, I still see places with the vuln_discussion field. It's worth to double check that because there seems to have duplicated data there.
> > I see that many of the changes are only formatting options, if we can't avoid that, I suggest we do that in a different pull request... maybe having the script to detect if the text is actually different and only perform the change if there is any. Then as a follow up we can open another pull request that will just perform this type of formatting change.

I don't see this as something that blocks the PR to get merged. Specially because the changes are automatically populated and we are only interested if the content is accurate, the git history is **not** that valuable in this context.

> 
> For how I have decided that addressing the `vuldiscussion` vs `vuln_discussion` should be reserved for another PR. So I have dropped those commits from the PR.

Okay, sounds good to me.

ggbecker reviewed (approved):
If you don't plan to do any further changes in this PR considering all the previous comments, then I'm good with it. Considering there will be follow ups.

Mab879 commented:
I moved policy specific content updates to #12365.

marcusburghardt commented:
Thanks @Mab879 and @ggbecker
